spacy/language.py

```diff
@@ -1155,21 +1155,21 @@ def begin_training(
 
         DOCS: https://nightly.spacy.io/api/language#begin_training
         """
-        # TODO: throw warning when get_gold_tuples is provided instead of get_examples
         if get_examples is None:
-            get_examples = lambda: []
-        else:  # Populate vocab
-            if not hasattr(get_examples, "__call__"):
-                err = Errors.E930.format(name="Language", obj=type(get_examples))
+            doc = Doc(self.vocab, words=["This is an example sentence."])
```

Comment:
It kinda doesn't matter but the use of `words` with a single sentence string here may confuse people reading the code base... Maybe we should always default to nonsense here? Like `["x", "y", "z"]` or something. Also to make it clear that it's not language-specific and doesn't actually matter.

Another thing to consider: if `get_examples is None`, we could use the new logger to log debugging info?
```python
util.logger.debug("No get_examples callback provided to Language.begin_training, creating dummy examples")
```

(This is a regular Python logger, so the output will only be shown if the user explicitly enables `DEBUG` logging. Added this recently to replace some of the hacky logging/intrusive warnings. We're not using it much at the moment but we can going forward.)

Reply:
Ah, good points, thanks! (the `words` stuff is definitely not OK)